BSP/Simulator/Src/BSP_Timer.c

```diff
@@ -0,0 +1,73 @@
+#include "BSP_Timer.h"
+
+static void (*function1);
+static void (*function2);
+int row=0; //keeps track of row number in CSV file
+/** 
+
+ * @brief   Updates the time by reading the CSV file and calling relevant functions at the right time 
+
+ * @param   none
+
+ * @return  none
+
+ */  
+
+void BSP_Timer_Update()
+{
+    FILE *file = fopen("Timer.csv", "r");
+    char arr[300];
+
+    int counter=0;
+
+    int current1;
+
+    int current2;
+
+
+    while(fgets(arr,sizeof(arr),file))
+    {
+        counter++;
+        if(counter == row)
+        {
+            fscanf(file,"%x , %x",&current1,&current2);

```

Comment:
The format is as follows:

2 lines in the CSV file

CURRENT1, RELOAD1
CURRENT2, RELOAD2

You won't need a row variable. Just read all rows (2) and store the values properly